core/src/main/java/org/apache/calcite/config/CalciteConnectionProperty.java

```diff
@@ -105,6 +105,10 @@
    * {@link org.apache.calcite.sql.parser.SqlParserImplFactory}. */
   PARSER_FACTORY("parserFactory", Type.PLUGIN, null, false),
 
+  /** Specifies whether to use top down optimization.
+   * {@link org.apache.calcite.plan.volcano.VolcanoPlanner#setTopDownOpt(boolean)} */
+  TOP_DOWN_OPT("topDownOpt", Type.BOOLEAN, true, false),
```

Comment:
I think you need a rebase to master. 

The similar code was already added in https://github.com/apache/calcite/pull/1985

Reply:
you are right. Thanks for the reminding

Reply:
@amaliujia, I noticed that in #1985, default value of calcite.planner.topdown.opt  has been changed to "false". Why we need this change?

Reply:
To be prepared for the top-down rule apply. With top-down rule apply, the original code path will be different. And there will be more plan diff combining with project/filter/join's trait propagation.

Reply:
Yes, Like what Haisheng said, enable top-down opt will generate different plans for existing tests. As we still under fast development for top-down opt, we thought it could be good to disable it globally for now. We will tackle those affected tests before fully enable top-down opt.

Reply:
IMHO, if we ignore those plan changes every time we commit code, the code is more vulnerable to bugs. And it will be even harder to fix plan changes in the future as we don't know which and whose commit introduced the plan change.

Reply:
In fact, I think Haisheng  was thinking that allows each test has two expected plans: one for top-down opt, and another for old planner. 

I think what you are saying is to make Volcano planner be fully compatible with top-down optimization. I think people were discussing whether there should be a new planner, or using existing Volcano planner to develop this top-down optimization.  I actually don't know if there was already an agreement of it or not.

Reply:
Two expected plans looks like a good solution.

There are discussion about keeping one planner versus creating a new one. But no matter which way to go, the volcano model will be deprecated some day. So compatibility is necessary for both alternatives. Plan could be changed, but the new model should not produce a worse plan.

Reply:
I tried to reopen the test cases for top-down (see my second commit). Looks like rebasing your changes does not introduce more plan changes on old cases.